driver-core/src/test/java/com/datastax/driver/core/ShardAwareTest.java

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright ScyllaDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.driver.core;
+
+import static com.datastax.driver.core.Assertions.assertThat;
+import static com.datastax.driver.core.CreateCCM.TestMode.PER_CLASS;
+
+import com.datastax.driver.core.QueryTrace.Event;
+import com.datastax.driver.core.policies.ConstantReconnectionPolicy;
+import com.datastax.driver.core.policies.LoadBalancingPolicy;
+import com.datastax.driver.core.policies.ReconnectionPolicy;
+import com.datastax.driver.core.policies.RoundRobinPolicy;
+import com.datastax.driver.core.policies.TokenAwarePolicy;
+import java.util.function.Predicate;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testng.annotations.Test;
+
+@CreateCCM(PER_CLASS)
+@CCMConfig(numberOfNodes = 3, config = "ring_delay_ms:10000")
+public class ShardAwareTest extends CCMTestsSupport {
+
+  static final Logger logger = LoggerFactory.getLogger(ShardAwareTest.class);
+
+  static void createKeyspaceAndColumnFamily(Session session) {
+    session.execute("DROP KEYSPACE IF EXISTS preparedtests");
+    session.execute(
+        "CREATE KEYSPACE preparedtests WITH replication = {'class': 'SimpleStrategy', 'replication_factor': '3'}");
+    session.execute("USE preparedtests");
+    session.execute("CREATE TABLE cf0 (a text, b text, c text, PRIMARY KEY (a, b))");
+  }
+
+  void createData(Session session) {
+    session.execute("USE preparedtests");
+    PreparedStatement prepared = session.prepare("INSERT INTO cf0 (a, b, c) VALUES  (?, ?, ?)");
+
+    BoundStatement bs1 = prepared.bind();
+    bs1.setString("a", "a");
+    bs1.setString("b", "b");
+    bs1.setString("c", "c");
+    session.execute(bs1);
```

Comment:
It seems that a little helper method would be useful not to duplicate this code 3 times.